src/Npgsql/CrateDbDatabaseInfo.cs

```diff
@@ -0,0 +1,235 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Npgsql.PostgresTypes;
+using Npgsql.TypeHandlers;
+using Npgsql.TypeHandlers.DateTimeHandlers;
+using Npgsql.TypeMapping;
+using NpgsqlTypes;
+
+namespace Npgsql
+{
+    class CrateDbDatabaseInfoFactory : INpgsqlDatabaseInfoFactory
+    {
+        /// <summary>
+        /// Returns a CrateDbDatabaseInfo instance if the given connection connects to a CrateDB cluster and null otherwhise.
+        /// </summary>
+        /// <param name="conn"></param>
+        /// <param name="timeout"></param>
+        /// <param name="async"></param>
+        /// <returns></returns>
+        public Task<NpgsqlDatabaseInfo> Load(NpgsqlConnection conn, NpgsqlTimeout timeout, bool async)
+              => Task.FromResult(
+                    conn.PostgresParameters.ContainsKey("crate_version")
+                          ? (NpgsqlDatabaseInfo)new CrateDbDatabaseInfo(conn)
+                          : null
+                 );
+    }
+
+    /// <summary>
+    /// Represents a NpgsqlDatabaseInfo-class for usage with CrateDB.
+    /// </summary>
+    public class CrateDbDatabaseInfo : NpgsqlDatabaseInfo
+    {
+        static readonly IDictionary<string, uint> CrateDbBaseTypes = new Dictionary<string, uint>
+        {
+            { "bool", 16 },
+            { "int2", 21 },
+            { "int4", 23 },
+            { "json", 114 },
+            { "float4", 700 },
+            { "char", 18 },
+            { "int8", 20 },
+            { "timestamptz", 1184 },
+            { "float8", 701 },
+            { "varchar", 1043 }
+        };
+
+        static readonly IDictionary<string, uint> CrateDbArrayTypes = new Dictionary<string, uint>
+        {
+            { "_bool", 1000 },
+            { "_int2", 1005 },
+            { "_int4", 1007 },
+            { "_json", 199 },
+            { "_float4", 1021 },
+            { "_char", 1002 },
+            { "_int8", 1016 },
+            { "_timestamptz", 1185 },
+            { "_float8", 1022 },
+            { "_varchar", 1015 }
+        };
+
+        static readonly IDictionary<string, string> PgTypeNameToInternalName = new Dictionary<string, string>
```

Comment:
Should this include a case-insensitive comparers?

Reply:
hi austindenski, thanks for your review. no there is no need for a case insensitive comparer here.